mdtraj/formats/pdb/pdbstructure.py

```diff
@@ -46,6 +46,98 @@
+    """
+    if pdbstructure._current_model is None or pdbstructure._current_model._current_chain is None or pdbstructure._current_model._current_chain._current_residue is None:
+        # This is the first residue in the model.
+        return pdbstructure._next_residue_number
+    else:
+        currentRes = pdbstructure._current_model._current_chain._current_residue
+        if currentRes.name_with_spaces != curr_atom.residue_name_with_spaces:
+            # The residue name has changed.
+            return pdbstructure._next_residue_number
+        elif curr_atom.name_with_spaces in currentRes.atoms_by_name:
+            # There is already an atom with this name.
+            return pdbstructure._next_residue_number
+        else:
+            return currentRes.number
+
+
+def _read_residue_number(num_str, pdbstructure=None, index_fnc=None, curr_atom=None):
+    """
+    Try to check what the residue number is.
+    """
+    try:
+        if num_str in _residue_num_initial_nodec_vals.keys() or pdbstructure._next_residue_number > 9999:
+            if num_str == "9999":
+                # If on the cusp... move on...
+                return int(num_str)
+            else:
+                # This is all the cases where we're safely in the hex/chimera/overflow region
+                raise OverflowError("Need to parse residue number using non-decimal residue modes.")
+        else:
+            #  Within "normal" pdb specifications
+            return int(num_str)
+    except (AttributeError, OverflowError, KeyError):
+        if index_fnc is None:
+            # we need to figure out on the 1st try which mode to switch to. There are currently 3 options: VMD (hex) and Chimera (their own 'hybrid36' mode) and Overflow (****).
+            # Chimera starts with A000, vmd with 2710, and Overflow just shows ****.
+            # The can be turned into decimal with "int()" so the "hex" mode will only be activated when _next_residue_number > 9999 (maximum in decimal) and current num_str isn't 9999.
+            if pdbstructure is None:
+                try:
+                    return int(num_str)
+                except ValueError:
+                    return 0  # num_str is not decimal, no index_fnc to interpret it, no pdbstructure to say what it is or to provide current number of atoms. No way to figure out
+            else:
+                if pdbstructure._residue_num_nondec_mode is None:
+                    pdbstructure._residue_num_nondec_mode = pdbstructure._residue_num_initial_nodec_vals[num_str]  # convert to correct mode
+
+                try:
+                    # Try and run the _residue_num_fncs
+                    return pdbstructure._residue_num_fncs[pdbstructure._residue_num_nondec_mode](num_str)
+                except ValueError:
+                    # Didn't work, we need to change to overflow mode and guess with _next_residue_number
+                    pdbstructure._atom_num_nondec_mode = "overflow"
```

Comment:
I think that here `_residue_num_nondec_mode` attribute should be changed

Reply:
Nice Catch. You're correct.